spec/handlers/sessions_create_spec.rb

```diff
@@ -356,6 +356,70 @@
             end
           end
         end
+
+        context "linked to no one" do
+          context "but user already has already authenticated with the same provider before" do
+            context "new_social user" do
+              let!(:user) { FactoryGirl.create(:new_social_user) }
+              let!(:authentication) { FactoryGirl.create(:authentication, user: user) }
+
+              it "returns a status output instead of raising an exception" do
+                result = lambda { |uid|
+                            described_class.handle(
+                              user_state: user_state,
+                              request: MockOmniauthRequest.new(authentication.provider, uid, {})
+                            )
+                          }
+
+                result.call(111)
+
+                same_provider_different_uid = result.call(222)
+                expect{same_provider_different_uid}.not_to raise_exception
```

Comment:
Did you ever get this line to fail?  I don't think it would, since `same_provider_different_uid` is not itself something that is being called (it is just an object).  Would have expected to see something like...

```
same_provider_different_uid = nil
expect{same_provider_different_uid = result.call(222)}.not_to raise_exception
```

By the way you can also put these three repeated lines in their own method call that just takes the `222` parameter.